huaweicloud/services/apig/resource_huaweicloud_apig_instance_routes.go

```diff
@@ -0,0 +1,173 @@
+		UpdateContext: resourceInstanceRoutesUpdate,
+		DeleteContext: resourceInstanceRoutesDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceInstanceRoutesImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The region where the dedicated instance and routes are located.",
+			},
+			"instance_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The ID of the dedicated instance to which the routes belong.",
+			},
+			"next_hops": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "The configuration of the next hop routes.",
+			},
+		},
+	}
+}
+
+func modifyInstanceRoutes(client *golangsdk.ServiceClient, instanceId string, routes []interface{}) error {
+	routeConfig := map[string]interface{}{
+		"user_routes": routes,
+	}
+	routeBytes, err := json.Marshal(routeConfig)
+	if err != nil {
+		return fmt.Errorf("error parsing routes configuration: %s", err)
+	}
+	opts := instances.FeatureOpts{
+		Name:   "route",
+		Enable: utils.Bool(true),
+		Config: string(routeBytes),
+	}
+	log.Printf("[DEBUG] The modify options of the instance routes is: %#v", opts)
+	_, err = instances.UpdateFeature(client, instanceId, opts)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func resourceInstanceRoutesCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	client, err := cfg.ApigV2Client(cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating APIG V2 client: %s", err)
+	}
+
+	var (
+		instanceId = d.Get("instance_id").(string)
+		routes     = d.Get("next_hops").(*schema.Set)
+	)
+	if err := modifyInstanceRoutes(client, instanceId, routes.List()); err != nil {
+		return diag.Errorf("error creating instance routes: %v", err)
+	}
+	d.SetId(instanceId)
+
+	return resourceInstanceRoutesRead(ctx, d, meta)
+}
+
+func resourceInstanceRoutesRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	client, err := cfg.ApigV2Client(cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating APIG V2 client: %s", err)
+	}
+
+	instanceId := d.Get("instance_id").(string)
+	opts := instances.ListFeaturesOpts{
+		// Default value of parameter 'limit' is 20, parameter 'offset' is an invalid parameter.
+		// If we omit it, we can only obtain 20 features, other features will be lost.
+		Limit: 500,
+	}
+	resp, err := instances.ListFeatures(client, instanceId, opts)
+	if err != nil {
+		return diag.Errorf("error querying feature list: %s", err)
+	}
+	log.Printf("[DEBUG] The feature list is: %v", resp)
+
+	var routeConfig string
+	for _, val := range resp {
+		if val.Name == "route" {
+			routeConfig = val.Config
```

Comment:
it is better to add a break to jump out the loop

Reply:
OFC